Test failures with sklearn 1.6

Only runs that pass are those with the oldest version of sklearn we support: all the others were run with 1.6

https://github.com/nilearn/nilearn/actions/runs/12241078484

Failures are due to some of the estimator checks

```
FAILED nilearn/maskers/tests/test_base_masker.py::test_check_estimator_invalid[estimator4-check4-check_positive_only_tag_during_fit]
FAILED nilearn/decomposition/tests/test_base.py::test_check_estimator[estimator9-check9-check_estimator_sparse_array] - AssertionError: Estimator _BaseDecomposition doesn't seem to fail gracefully on sparse data: error message should state explicitly that sparse input is not supported if this is not the case, e.g. by using check_array(X, accept_sparse=False).
FAILED nilearn/maskers/tests/test_multi_nifti_masker.py::test_check_estimator[estimator7-check7-check_estimator_sparse_array] - AssertionError: Estimator MultiNiftiMasker doesn't seem to fail gracefully on sparse data: error message should state explicitly that sparse input is not supported if this is not the case, e.g. by using check_array(X, accept_sparse=False).
FAILED nilearn/maskers/tests/test_multi_nifti_masker.py::test_check_estimator[estimator19-check19-check_estimator_sparse_array] - AssertionError: Estimator NiftiMasker doesn't seem to fail gracefully on sparse data: error message should state explicitly that sparse input is not supported if this is not the case, e.g. by using check_array(X, accept_sparse=False).
FAILED nilearn/regions/tests/test_hierarchical_kmeans_clustering.py::test_check_estimator_invalid[estimator2-check2-check_positive_only_tag_during_fit]
````

--- a/nilearn/decomposition/tests/test_base.py
+++ b/nilearn/decomposition/tests/test_base.py
@@ -3,9 +3,7 @@ import pytest
 from nibabel import Nifti1Image
 from numpy.testing import assert_array_almost_equal
 from scipy import linalg
-from sklearn import __version__ as sklearn_version
 
-from nilearn._utils import compare_version
 from nilearn._utils.class_inspect import check_estimator
 from nilearn.conftest import _affine_eye, _img_3d_ones, _rng
 from nilearn.decomposition._base import (
@@ -23,8 +21,6 @@ extra_valid_checks = [
     "check_transformers_unfitted",
     "check_transformer_n_iter",
 ]
-if compare_version(sklearn_version, ">=", "1.6"):
-    extra_valid_checks.append("check_estimator_sparse_array")
 
 
 @pytest.mark.parametrize(

--- a/nilearn/maskers/tests/test_base_masker.py
+++ b/nilearn/maskers/tests/test_base_masker.py
@@ -31,7 +31,7 @@ extra_valid_checks = [
 if compare_version(sklearn_version, "<", "1.5.0"):
     extra_valid_checks.append("check_estimator_sparse_data")
 
-if compare_version(sklearn_version, ">", "1.6"):
+if compare_version(sklearn_version, ">", "1.5.2"):
     extra_valid_checks.append("check_positive_only_tag_during_fit")
 
 

--- a/nilearn/maskers/tests/test_multi_nifti_masker.py
+++ b/nilearn/maskers/tests/test_multi_nifti_masker.py
@@ -9,9 +9,7 @@ import pytest
 from joblib import Memory
 from nibabel import Nifti1Image
 from numpy.testing import assert_array_equal
-from sklearn import __version__ as sklearn_version
 
-from nilearn._utils import compare_version
 from nilearn._utils.class_inspect import check_estimator
 from nilearn._utils.exceptions import DimensionError
 from nilearn._utils.testing import write_imgs_to_path
@@ -24,9 +22,6 @@ extra_valid_checks = [
     "check_transformer_n_iter",
     "check_transformers_unfitted",
 ]
-
-if compare_version(sklearn_version, ">=", "1.6"):
-    extra_valid_checks.append("check_estimator_sparse_array")
 
 
 @pytest.mark.parametrize(

--- a/nilearn/regions/tests/test_hierarchical_kmeans_clustering.py
+++ b/nilearn/regions/tests/test_hierarchical_kmeans_clustering.py
@@ -50,7 +50,7 @@ if compare_version(sklearn_version, ">", "1.5.2"):
 if compare_version(sklearn_version, "<", "1.5.0"):
     extra_valid_checks.append("check_estimator_sparse_data")
 
-if compare_version(sklearn_version, ">", "1.6"):
+if compare_version(sklearn_version, ">=", "1.6"):
     extra_valid_checks.append("check_positive_only_tag_during_fit")
 
 

